src/App.tsx

```diff
@@ -1,27 +1,72 @@
-import React from 'react';
+import React, { useState } from 'react';
+
 import './App.scss';
-import { GoodsList } from './GoodsList';
+import { GoodsList } from './components/GoodList/GoodsList';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
+import { Loader } from './components/Loader/Loader';
+
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleGoods = (getGoods: () => Promise<Good[]>) => {
+    setIsLoading(true);
+
+    getGoods()
+      .then(setGoods)
+      .catch(() => setErrorMessage('Try again later'))
+      .finally(() => setIsLoading(false));
+  };
+
+  const renderContent = () => {
+    if (isLoading) {
+      return <></>;
+    }
```

Comment:
why do you need an empty jsx during the loading?
just show loading or GoodsList in jsx

also, check error below returns error message if there is some error, so it's okay